sdk/servicebus/Microsoft.Azure.ServiceBus/tests/TokenProviderTests.cs

```diff
@@ -61,13 +33,98 @@ public async Task SasTokenWithLargeExpiryTimeShouldBeAccepted()
 
                 try
                 {
-                    var msg = await receiver.ReceiveAsync(TimeSpan.FromSeconds(5));
+                    var msg = await receiver.ReceiveAsync(TimeSpan.FromSeconds(5)).ConfigureAwait(false);
                 }
                 finally
                 {
-                    await receiver.CloseAsync();
+                    await receiver.CloseAsync().ConfigureAwait(false);
                 }
             });
         }
+
+        /// <summary>
+        /// This test is for manual only purpose.
+        /// </summary>
+        [Fact]
+        [LiveTest]
```

Comment:
>LiveTest [](start = 9, length = 8)

FMU, What does this do? 
Does it ensure it doesn't run on CI system?

Reply:
I think this has to do with the "Live Unit Testing" feature from Visual Studio, this LiveTest attribute indicates that this test will be run if live testing is turned on.

No particular reason why I used it, I just see that every single test has it, so I just copy pasted it anyways.

---
In reply to: [286702729](https://github.com/Azure/azure-sdk-for-net/pull/6393#discussion_r286702729) [](ancestors = 286702729)

Reply:
`LiveTest` is an attribute applying a `TestCategory` trait that is used by the engineering system to denote that a given test depends on "Live" Azure services (as opposed to a self-contained) unit test.    _(see [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/tests/Infrastructure/LiveTestAttribute.cs))_

Please use it only for tests that must run against Azure.

Reply:
That's good to know. @jsquire, your input might be required here to convert this to a good live test. What's the best way to get the tenantId/appId etc.

---
In reply to: [286713480](https://github.com/Azure/azure-sdk-for-net/pull/6393#discussion_r286713480) [](ancestors = 286713480)

Reply:
I'd suggest through environment variables, which is how our engineering system prefers to expose secrets.  Those would likely be held in KeyVault or an environmental configuration, depending on the sensitivity of what is being stored.  If I recall, there's a similar environment read exposed through `TestUtilities` for the connection string.

You'll also need to update the ARM template to create the needed service principal and any other scaffolding.   The goal would still be to make this something that contributors could pull down, build, and run tests from with minimal hassle.

I'd first wonder if there is a reasonable  way to mock rather than making it a live test.   This is a file about tests for the token provider, and this test is exercising much more than the token provider;  it's really an end-to-end integration test for the token provider and RBAC in the serivce.